### PR TITLE
VDS: allow for specifying an arbitrary secrets path

### PR DESCRIPTION
**Pull request by benashz**
This PR ensure that VDS can support any Vault secret engine path.

- drops hard coded `creds` path component.

Example of an updated VDS CR:
```yaml
apiVersion: secrets.hashicorp.com/v1alpha1
kind: VaultDynamicSecret
metadata:
  name: vso-db-demo-create
  namespace: demo-ns
spec:
  destination:
    create: true
    name: vso-db-demo-created
  mount: demo-db
  path: creds/dev-postgres
  rolloutRestartTargets:
  - kind: Deployment
    name: vso-db-demo
```

**Breaking change:**
- Spec.Role is now Spec.Path

Close #108 #107

### CODE REVIEW

File: demo/infra/app/variables.tf [hunk 1]
```diff
@@ -30,5 +30,9 @@ variable "k8s_db_secret_count" {
   default = 50
 }
 
+variable "db_role" {
```
Comment (kschoche) [inline demo/infra/app/variables.tf:33]: Not 100% on this, but I think there is a `local.db_role` which is now unused due to this change that can be removed?

**Review by kschoche — APPROVED** [review]
This looks great, I left a couple smallish comments that I'd like to circle back on before merging if you get a chance.
Also a reminder to add a changelog entry (especially since this is a breaking change). I think it would be helpful to disclose the spec change and how to migrate in that entry.

File: demo/infra/app/variables.tf (hunk 1 above)
Comment (benashz) [reply demo/infra/app/variables.tf:33]: Oops, need to drop `local.db_role`

**Comment by benashz** [comment]
> This looks great, I left a couple smallish comments that I'd like to circle back on before merging if you get a chance. Also a reminder to add a changelog entry (especially since this is a breaking change). I think it would be helpful to disclose the spec change and how to migrate in that entry.

@kschoche  Good call on the changelog updates. For the breaking changes + migration instructions, I was thinking we could do that as a part of the upcoming release, since I think we have a few others in the works?

**Comment by kschoche** [comment]
> > This looks great, I left a couple smallish comments that I'd like to circle back on before merging if you get a chance. Also a reminder to add a changelog entry (especially since this is a breaking change). I think it would be helpful to disclose the spec change and how to migrate in that entry.
> 
> @kschoche Good call on the changelog updates. For the breaking changes + migration instructions, I was thinking we could do that as a part of the upcoming release, since I think we have a few others in the works?

I think that sounds fair, I'd still prefer to have breaking changes noted in the changelog + duplicated in release notes so users can get that info directly from the changelog. Migration steps could be elsewhere indeed.

**Comment by benashz** [comment]
> 



> > > This looks great, I left a couple smallish comments that I'd like to circle back on before merging if you get a chance. Also a reminder to add a changelog entry (especially since this is a breaking change). I think it would be helpful to disclose the spec change and how to migrate in that entry.
> > 
> > 
> > @kschoche Good call on the changelog updates. For the breaking changes + migration instructions, I was thinking we could do that as a part of the upcoming release, since I think we have a few others in the works?
> 
> I think that sounds fair, I'd still prefer to have breaking changes noted in the changelog + duplicated in release notes so users can get that info directly from the changelog. Migration steps could be elsewhere indeed.

Yeah, I was only referring to the instruction bit.

**Review by kschoche — APPROVED** [review]
Looks great!

**Pull request merged**